custom/icds_reports/views.py

```diff
@@ -1569,15 +1557,19 @@ def get(self, request, *args, **kwargs):
 
 
 @method_decorator([login_and_domain_required], name='dispatch')
-class CheckPDFReportStatus(View):
+class CheckExportReportStatus(View):
     def get(self, request, *args, **kwargs):
         task_id = self.request.GET.get('task_id', None)
+        is_issnip_monthly_register = self.request.GET.get('is_issnip_monthly_register', None)
 
         res = AsyncResult(task_id)
         status = res.ready()
 
         if status:
-            task_result = prepare_issnip_monthly_register_reports.AsyncResult(task_id)
+            if is_issnip_monthly_register:
+                task_result = prepare_issnip_monthly_register_reports.AsyncResult(task_id)
```

Comment:
Is it possible to import `AsyncResult` from celery, so that another parameter doesn't need to be passed? http://docs.celeryproject.org/en/3.1/reference/celery.result.html#celery.result.AsyncResult